infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/provision/OpenShiftStopWorkspaceRoleProvisioner.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.openshift.provision;
+
+import com.google.inject.Inject;
+import io.fabric8.kubernetes.api.model.ObjectReferenceBuilder;
+import io.fabric8.openshift.api.model.*;
+import io.fabric8.openshift.client.OpenShiftClient;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.workspace.infrastructure.openshift.OpenShiftClientFactory;
+import org.eclipse.che.workspace.infrastructure.openshift.environment.OpenShiftCheInstallationLocation;
+
+public class OpenShiftStopWorkspaceRoleProvisioner {
+
+  private final OpenShiftClientFactory clientFactory;
+  private final OpenShiftCheInstallationLocation installationLocation;
+
+  @Inject
+  public OpenShiftStopWorkspaceRoleProvisioner(
+      OpenShiftClientFactory clientFactory, OpenShiftCheInstallationLocation installationLocation) {
+    this.clientFactory = clientFactory;
+    this.installationLocation = installationLocation;
+  }
+
+  public void provision(String projectName) throws InfrastructureException {
```

Comment:
technically speaking `installationLocation` can be null based on implementation, so in this case can we log error and avoid doing anything in the provision method?

Reply:
@tomgeorge can we check `getInstallationLocationNamespace` here and do nothing (log error) if null?